src/Native/Unix/configure.cmake

```diff
@@ -321,6 +321,85 @@ check_function_exists(
     mach_timebase_info
     HAVE_MACH_TIMEBASE_INFO)
 
+set (PREVIOUS_CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS})
+set (CMAKE_REQUIRED_FLAGS "-Werror -Wsign-conversion")
+
+check_cxx_source_compiles(
+    "
+    #include <sys/socket.h>
+
+    int main()
+    {
+        int fd;
+        sockaddr* addr;
+        socklen_t addrLen;
+
+        int err = bind(fd, addr, addrLen);
+        return 0;
+    }
+    "
+    BIND_ADDRLEN_UNSIGNED
+)
+
+check_cxx_source_compiles(
+    "
+    #include <sys/socket.h>
+
+    int main()
+    {
+        int fd;
+        sockaddr* addr;
+        int32_t addrLen;
+
+        int err = bind(fd, addr, addrLen);
+        return 0;
+    }
+    "
+    BIND_ADDRLEN_SIGNED
+)
+
+if(NOT BIND_ADDRLEN_UNSIGNED AND NOT BIND_ADDRLEN_SIGNED)
+    message(FATAL_ERROR "The addrlen parameter in bind() must be a signed int or an unsigned int.")
+endif()
```

Comment:
That's fair.  It just seems like overkill for now, especially since I expect there are a bunch of other places we'd need to change to match this if that was the approach we wanted to take.  We're not particularly consistent right now.  I'd err for now on the side of not having these extra checks, and then if in the future we wanted to follow your suggested approach, make a pass through doing it for all places we currently have if/else checks for compilation constants.